### PR TITLE
APEXMALHAR-2291 Fix for Exactly-once processing of JdbcPOJOInsertOutput Operator.

### PR DESCRIPTION
**Pull request by KapoorHitesh**
Added a check in endWindow() to not to commit if committed window id is greater than current window id.
@bhupeshchawda @chinmaykolhatkar Please review. Have done the code changes as discussed on mail thread, travis build has passed, have written corresponding unit test case

### CODE REVIEW

**Comment by tweise** [comment]
Please update with a more descriptive commit message.

File: library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoOperatorTest.java [hunk 1]
```diff
@@ -335,6 +335,90 @@ public void testJdbcPojoInsertOutputOperator()
...
+
+    outputOperator.beginWindow(2);
+    for (int i = 50; i < 60; i++) {
+      outputOperator.input.process(events.get(i));
+    }
+    outputOperator.endWindow();
+
+    Assert.assertEquals("rows in db", 30, outputOperator.getNumOfEventsInStore(TABLE_POJO_NAME));
+
+  }
```
Comment (bhupeshchawda) [inline library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoOperatorTest.java:426]: Can you try to add the following scenario:
Window 0 = 0-9
Window 1 = 10-19
Window 2 = 20-29
Operator Crash
Window 0 = 0-9
Window 1 = 10-19
Window 2 = 20-29
Window 3 = 30-39

Validate that total number of rows = 40

**Review by bhupeshchawda — COMMENTED** [review]
LGTM. Can you just address the test case scenario?

File: library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoOperatorTest.java [hunk 2]
```diff
@@ -335,6 +335,95 @@ public void testJdbcPojoInsertOutputOperator()
...
+    for (int i = 50; i < 60; i++) {
+      outputOperator.input.process(events.get(i));
+    }
+
+    outputOperator.beginWindow(3);
+    for (int i = 60; i < 70; i++) {
+      outputOperator.input.process(events.get(i));
+    }
+    outputOperator.endWindow();
+
```
Comment (bhupeshchawda) [inline library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoOperatorTest.java:420]: Add deactivate() and teardown()

Comment (KapoorHitesh) [reply library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcPojoOperatorTest.java:420]: Thanks. Done.

**Pull request merged**